aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -135,15 +136,29 @@ Tensor angle(const Tensor& self) {
 }
 
 Tensor real(const Tensor& self) {
-  TORCH_CHECK(!self.is_complex(), "real is not yet implemented for complex tensors.");
-  return self;
+  if (self.is_complex()) {
+    auto float_tensor = at::native::view_complex_as_float(self);
+    if (self.numel() == 0) {
+      return float_tensor;
+    } else {
+      return at::select(float_tensor, float_tensor.dim() - 1, 0);
+    }
+  } else {
+    TORCH_CHECK(false, "real is not implemented for tensors with non-complex dtypes.");
+  }
 }
 
 Tensor imag(const Tensor& self) {
-  TORCH_CHECK(false, "imag is not yet implemented.");
-
-  // Note: unreachable
-  return at::zeros_like(self);
+  if (self.is_complex()) {
+    auto float_tensor = at::native::view_complex_as_float(self);
+    if (self.numel() == 0) {
```

Comment:
This special case looks strange!

Reply:
yeah good catch! removed the special case since it was unnecessary and also inconsistent with the definition where we state that `view_complex_as_float` returns a tensor of size `m1 x m2 x ... mn x 2` for a tensor of size `m1 x m2 x ... mn` . It should remain the case for tensors with zero elements as well.